src/main/java/gui/JStatGuiApplication.java

```diff
@@ -0,0 +1,43 @@
+package gui;
+
+import detail.JStatGuiGlobalData;
+import detail.MapDataContainer;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+import java.util.concurrent.Executors;
+
+
+/**
+ * We ar running the application from an executable JAR. Thus,  it’s important to
+ * have a main class that will be executed when that JAR file is run. We also need at
+ * least a minimal amount of Spring configuration to bootstrap the application. 
+ *
+ */
+@SpringBootApplication
+public class JStatGuiApplication {
+
+    public static void main(String[] args) {
+
+        // do the needed initializations
```

Comment:
do the initialisations need to be done on application launch?

Reply:
I can't see any other point  doing so. If not in the main then we will end up with if statements all around...

Reply:
agreed!